crypto/merkle/simple_map.go

```diff
@@ -21,13 +24,13 @@ func newSimpleMap() *simpleMap {
 }
 
 // Set hashes the key and value and appends it to the kv pairs.
```

Comment:
Looking at the code, this comment seems outdated. Only the value seems to be hashed?